lib/AST/NameLookup.cpp

```diff
@@ -3875,20 +3876,45 @@ void FindLocalVal::visitBraceStmt(BraceStmt *S, bool isTopLevelCode) {
       return;
   }
 
+  // Visit inner statements first before reporting local decls in the current
+  // scope.
   for (auto elem : S->getElements()) {
     // If we have a SingleValueStmtExpr, there may be local bindings in the
     // wrapped statement.
     if (auto *E = elem.dyn_cast<Expr *>()) {
       if (auto *SVE = dyn_cast<SingleValueStmtExpr>(E))
         visit(SVE->getStmt());
+      continue;
     }
+
     if (auto *S = elem.dyn_cast<Stmt*>())
       visit(S);
```

Comment:
Very nit-picky: Wouldn’t this be cleaner represented as

```suggestion
    } else if (auto *S = elem.dyn_cast<Stmt*>()) {
      visit(S);
    }
```

Reply:
I actually don't like `else if` branches for iterating union cases, unless the branch contents are almost the same or one line code. For example, this is OK,
```cpp
void ASTNode::walk(ASTWalker &Walker) {
  if (auto *E = this->dyn_cast<Expr*>())
    E->walk(Walker);
  else if (auto *S = this->dyn_cast<Stmt*>())
    S->walk(Walker);
  else if (auto *D = this->dyn_cast<Decl*>())
    D->walk(Walker);
  else if ...
```
But for this specific case I prefer `if ... continue`.

Reply:
OK, if you’ve got a preference, that’s fine with me.